File: Compare Version Numbers.cpp
// This is the single char version
vector<string> explode(const string &str, char delim){
    string temp;
    vector<string> res;
    for (size_t i = 0; i < str.size(); i++){
        if (str[i] == delim){
            if (!temp.empty()) res.push_back(temp);
            temp.clear();
        }
        else if (i == str.size()-1){
            temp += str[i];
            res.push_back(temp);
        }
        else temp += str[i];
    }
    return res;
}

class Solution {
public:
    int compareVersion(string version1, string version2) {
        vector<string> v1 = explode(version1, '.');
        vector<string> v2 = explode(version2, '.');
        int i = 0;
        while (1){
            if (i >= v1.size() || i >= v2.size()){
                if (v1.size() < v2.size()){
                    for (; i < v2.size(); i++)
                        if (stoi(v2[i])) return -1;
                    return 0;
                }
                else if (v1.size() > v2.size()){
                    for (; i < v1.size(); i++)
                        if (stoi(v1[i])) return 1;
                    return 0;
                }
                else return 0;
            }
            if (stoi(v1[i]) > stoi(v2[i])) return 1;
            else if (stoi(v1[i]) < stoi(v2[i])) return -1;
            i++;
        }
        return 0;
    }
};

File: Compare Version Numbers.py
class Solution(object):
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        v1 = version1.split('.')
        v2 = version2.split('.')
        i = 0
        while True:
            if i >= len(v1) or i >= len(v2):
                if len(v1) < len(v2):
                    for i in range(i, len(v2)):
                        if int(v2[i]) != 0: return -1
                    return 0

                elif len(v1) > len(v2):
                    for i in range(i, len(v1)):
                        if int(v1[i]) != 0: return 1
                    return 0

                else:
                    return 0

            if int(v1[i]) > int(v2[i]): return 1
            elif int(v1[i]) < int(v2[i]): return -1
            i += 1

        return 0